internal/node/node.bzl

```diff
@@ -250,8 +250,10 @@ fi
     # when building the image as that will reflect the selected --platform.
     node_tool_files = ctx.files.node[:]
 
-    # this should be resolved the same as above
-    node_tool_files.extend(ctx.toolchains["@build_bazel_rules_nodejs//toolchains/node:toolchain_type"].nodeinfo.tool_files)
+    if ctx.attr.toolchain:
```

Comment:
can reduce duplication by assigning to a local variable that holds either ctx.attr.toolchain or ctx.toolchains[default]